From c207c8b8dfd2d6e73f2014da69bbe437e49f21db Mon Sep 17 00:00:00 2001
From: Marius Tomaschewski <mt@suse.de>
Date: Thu, 13 Oct 2022 14:57:15 +0200
Subject: [PATCH] dhcp6: refresh ipv6 flags on staring in auto mode

In some cases, like during an "ifup ; ifup" there is no NEWLINK
event from kernel providing the IPv6 ready, rs-sent, ra-rcvd,
ra-managed or ra-otherconf flags about the router advertisement
(RA) processing we're using to reliably resolve the auto mode
(aka follow-ra mode), so we've changed to actively query them.

(cherry picked from commit 3f1604e716729ae410ed64a9512a87e1868fcf34)
---
 src/dhcp6/device.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/dhcp6/device.c b/src/dhcp6/device.c
index 9b9bb009..f36b16c9 100644
--- a/src/dhcp6/device.c
+++ b/src/dhcp6/device.c
@@ -487,8 +487,7 @@ ni_dhcp6_device_start_auto_prefix(ni_dhcp6_device_t *dev)
 
 	/* refresh in case kernel forgot to send it
 	 * (we increment timeout between attempts) */
-	if (dev->config->dry_run != NI_DHCP6_RUN_NORMAL)
-		ni_dhcp6_device_refresh_mode(dev, ifp);
+	ni_dhcp6_device_refresh_mode(dev, ifp);
 
 	/* request prefix after 1/3 defer timeout */
 	ni_timer_get_time(&now);
@@ -524,8 +523,7 @@ ni_dhcp6_device_start_auto(ni_dhcp6_device_t *dev)
 
 	/* refresh in case kernel forgot to send it
 	 * (we increment timeout between attempts) */
-	if (dev->config->dry_run != NI_DHCP6_RUN_NORMAL)
-		ni_dhcp6_device_refresh_mode(dev, ifp);
+	ni_dhcp6_device_refresh_mode(dev, ifp);
 
 	if (dev->config->mode & NI_BIT(NI_DHCP6_MODE_AUTO))
 		return ni_dhcp6_device_start_timer_arm(dev);
-- 
2.40.1

